cluster/store.go

```diff
@@ -179,17 +182,31 @@ type Store struct {
 }
 
 func NewFSM(cfg Config) Store {
-	return Store{
-		cfg:          cfg,
-		log:          cfg.Logger,
-		candidates:   make(map[string]string, cfg.BootstrapExpect),
-		applyTimeout: time.Second * 20,
-		raftResolver: resolver.NewRaft(resolver.RaftConfig{
+	// We have different resolver in raft so that depending on the environment we can resolve a node-id to an IP using
+	// different methods.
+	var raftResolver types.RaftResolver
+	if cfg.EnableFQDNResolver {
+		raftResolver = resolver.NewFQDN(resolver.FQDNConfig{
+			TLD:               cfg.FQDNResolverTLD,
+			RaftPort:          cfg.RaftPort,
+			IsLocalHost:       cfg.IsLocalHost,
+			NodeNameToPortMap: cfg.NodeNameToPortMap,
+		})
+	} else {
+		raftResolver = resolver.NewRaft(resolver.RaftConfig{
 			NodeToAddress:     cfg.NodeToAddressResolver,
 			RaftPort:          cfg.RaftPort,
 			IsLocalHost:       cfg.IsLocalHost,
 			NodeNameToPortMap: cfg.NodeNameToPortMap,
-		}),
+		})
```

Comment:
**suggestion:** maybe instead of if else, we set default to `cfg.NodeToAddressResolver` and replace it by `cfg.FQDNResolverTLD` if was enabled